### PR TITLE
Run SecOps demo with bounded memory.

### PR DESCRIPTION
**Pull request by ryzhyk**
This PR makes a series of changes that allow the SecOps demo to run indefinitely, while keeping it's memory usage bounded (~3.5GB on my system). See individual commits for details.

### CODE REVIEW

File: demo/project_demo00-SecOps/simulator/src/main.rs [hunk 1]
```diff
@@ -1,16 +1,20 @@
-use csv::WriterBuilder as CsvWriterBuilder;
 use futures::executor::block_on;
 use mockd::datetime::date_range;
 use rand::{random, thread_rng, Rng};
 use rdkafka::{
     admin::{AdminClient, AdminOptions, NewTopic, TopicReplication},
-    client::DefaultClientContext,
+    // client::DefaultClientContext,
```
Comment (blp) [inline demo/project_demo00-SecOps/simulator/src/main.rs:6]: I would just delete this.

File: demo/project_demo00-SecOps/simulator/src/main.rs [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
     config::{FromClientConfig, RDKafkaLogLevel},
     producer::{BaseRecord, DefaultProducerContext, Producer, ThreadedProducer},
     util::Timeout,
     ClientConfig,
 };
 use serde::Serialize;
+use serde_json::json;
+use std::{
+    sync::Arc,
+};
```
Comment (blp) [inline demo/project_demo00-SecOps/simulator/src/main.rs:16]: I think that rustfmt would condense this to `use std::sync::Arc`. Perhaps rustfmt has not been run.

File: demo/project_demo00-SecOps/simulator/src/main.rs
Comment (blp) [inline demo/project_demo00-SecOps/simulator/src/main.rs:1]: There's a lot of commented-out code in this file. I'd remove it.

File: demo/project_demo00-SecOps/simulator/src/main.rs [hunk 3]
```diff
@@ -94,32 +99,92 @@ impl KafkaProducer {
...
-        for batch in data {
-            let mut writer = CsvWriterBuilder::new()
-                .has_headers(false)
-                .from_writer(Vec::with_capacity(batch.len() * 32));
+        self.send_to_topic(topic, data.into_iter().map(|x| (x, true)), message_size)
+    }
 
-            for val in batch.iter().cloned() {
-                writer.serialize(val).unwrap();
+    pub fn send_to_topic<T, I>(&self, topic: &str, data: I, message_size: usize)
```
Comment (blp) [inline demo/project_demo00-SecOps/simulator/src/main.rs:113]: The naming is weird here. The name `message_size` implies (to my mind) that we're measuring the size of a message, but it's used as a maximum count. I'd either name it something like `max_updates` or change the send condition to something like `bytes.len() >= message_size`. (If you do the latter you don't need the `updates` variable anymore.)

**Review by blp — APPROVED** [review]
I read it and made a few comments.

File: python/feldera-api-client/feldera_api_client/models/json_encoder_config.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    buffer_size_records: Union[Unset, int] = UNSET
+    additional_properties: Dict[str, Any] = field(init=False, factory=dict)
+
+    def to_dict(self) -> Dict[str, Any]:
+        array = self.array
+        buffer_size_records = self.buffer_size_records
+
+        field_dict: Dict[str, Any] = {}
+        field_dict.update(self.additional_properties)
+        field_dict.update({})
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply python/feldera-api-client/feldera_api_client/models/json_encoder_config.py:28]: This is auto-generated code.  Sorry about the confusion.  I am thinking of abandoning the Python OpenAPI generator going forward.  Not sure it actually helps.

File: python/feldera-api-client/feldera_api_client/models/json_parser_config.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                # Example
+
+                ```json
+                [{"b": true, "i": 0},{"b": false, "i": 100, "s": "foo"}]
+                ```
+            update_format (Union[Unset, JsonUpdateFormat]): Supported JSON data change event formats.
+
+                Each element in a JSON-formatted input stream specifies
+                an update to one or more records in an input table.  We support
+                several different ways to represent such updates.
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply python/feldera-api-client/feldera_api_client/models/json_parser_config.py:58]: Same problem, auto-generated code.

File: demo/demo.sh [hunk 6]
```diff
@@ -4,15 +4,31 @@ THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
...
-set -e
-
 export REDPANDA_BROKERS=localhost:9092
 
-docker kill redpanda || true
-docker rm redpanda || true
-docker run --name redpanda -p 9092:9092 --rm -itd docker.redpanda.com/vectorized/redpanda:v23.1.13
+docker kill redpanda
+docker rm redpanda
+docker run --name redpanda -p 9092:9092 --rm -itd docker.redpanda.com/vectorized/redpanda:v23.2.3 || { echo 'Failed to start RedPanda container' ; exit 1; }
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply demo/demo.sh:11]: Agreed. I am hoping to ask @lalithsuresh to help me use our docker compose file instead here.

**Pull request merged**